contracts/TimeManagerV8.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import { SECONDS_PER_YEAR } from "./constants.sol";
+
+abstract contract TimeManagerV8 {
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable blocksOrSecondsPerYear;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    bool public immutable isTimeBased;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    function() view returns (uint256) private immutable _getCurrentSlot;
+
+    /// @notice Thrown when blocks per year is invalid
+    error InvalidBlocksPerYear();
+
+    /// @notice Thrown when time based but blocks per year is provided
+    error InvalidTimeBased();
+
+    /**
+     * @param timeBased_ A boolean indicating whether the contract is based on time or block.
+     * If timeBased is true than blocksPerYear_ param is ignored as blocksOrSecondsPerYear is set to SECONDS_PER_YEAR
+     * @param blocksPerYear_ The number of blocks per year
+     * @custom:error InvalidBlocksPerYear is thrown if blocksPerYear entered is zero
+     * @custom:oz-upgrades-unsafe-allow constructor
+     */
+    constructor(bool timeBased_, uint256 blocksPerYear_) {
+        if (!timeBased_ && blocksPerYear_ == 0) {
+            revert InvalidBlocksPerYear();
+        }
+
+        if (timeBased_ && blocksPerYear_ != 0) {
+            revert InvalidTimeBased();
```

Comment:
Should we rename the custom error to: `InvalidTimeBasedConfiguration`
for both versions of **TimeManager**.